### ISSUE TITLE
Unable save large file to S3 using Parse server

### ISSUE THREAD

**Issue by kesongxie** [ParsePlatform/parse-server#3532]
I tried to save to S3 bucket using Parse, and it can be saved correctly when the file is small, such as 864.2KB. However, when the file is large, say 5MB, it complaints with a message saying: "The data couldn’t be read because it isn’t in the correct format"

I'm using the following code to save the the video file to the S3

```
 func saveVideo(withVideoURL url: URL){
        let post = PFObject(className: "Post")
        post["caption"] = "Out of the game for 6 months, but back with vengeance. Meet your 2017 AO Men's champion"
        do{
            let data = try Data(contentsOf: url)
            print(data)
            post["media"] = PFFile(data: data)
            post.saveInBackground { (success, error) in
                if success{
                    print("video saved")
                }else{
                    print("failed")
                    if error != nil{
                        print(error!.localizedDescription)
                    }else{
                        print("erorr is nil")
                    }
                }
            }
        }catch let error as NSError{
            print("can't read")
            print(error.localizedDescription)
        }
    }
```

**Besides, even when the small video file is indeed being saved to the S3, it contains an extension .bin instead of, for example .mp4. I wonder what's happening here**
The url end up looking something like this
```
https://s3-us-west-1.amazonaws.com/sampleApp/19d5bce20f8b55te1b1b8f370212533e5_file.bin
```

**Comment by kesongxie** [comment]
Update: I have to use ` post["media"] = PFFile(data: data, contentType: "video/mp4")`, now the video is correctly stored in the S3 with mp4 extension, and I can also stream it using AVPlayer

**Comment by flovilmart** [comment]
Probably a problem with the load balancer in front of parse-server or the `maxUploadSize` option.

**Comment by flovilmart** [comment]
Probably a problem with your load balancer or the max

**Comment by kesongxie** [comment]
@flovilmart can you point me to any resources that I can verify the maxUploadSize. I understand the the default max upload size is MB, is that correct?

**Comment by flovilmart** [comment]
The default should be 20Mb, you're right, the option is passed directly to express body parser: https://github.com/ParsePlatform/parse-server/blob/d5940b17b6914f04462420812aed9efe54a2b37f/src/Routers/FilesRouter.js#L22

**Comment by kesongxie** [comment]
In this case, can I just set my environment variable the same as how I set the variable such as `MASTER_KEY` or `APP_ID`, etc, and I added a new environment variable called `maxUploadSize`with its value equals to `20mb`. Will this overwrite the default setting?

**Comment by flovilmart** [comment]
That should be possible     env: "PARSE_SERVER_MAX_UPLOAD_SIZE",

**Comment by kesongxie** [comment]
@flovilmart  Should the value be a string or a number? 20mb or 20?

**Comment by kesongxie** [comment]
Still complaining "The data couldn’t be read because it isn’t in the correct format." using saveInBackground when the file size is just 2.8MB

**Comment by flovilmart** [comment]
What is complaining? The client or the server?

**Comment by kesongxie** [comment]
The output from the Xcode console

**Comment by flovilmart** [comment]
so that's on your elastic beanstalk configuration then:

http://lmgtfy.com/?q=elastic+beanstalk+set+max+upload+size
http://serverfault.com/questions/618101/aws-elastic-beanstalk-file-upload-limit-issue#679854

**Comment by kesongxie** [comment]
Now it works after setting the client_max_body_size to a larger value. See how you set the value, you can refer it here http://websistent.com/fix-client-intended-to-send-too-large-body-nginx-error/

**Comment by HackShitUp** [comment]
@flovilmart could the default should be 20Mb file size be modified from the index.js file from the example parse server?

**Comment by flovilmart** [comment]
You should set it in your configuration object

**Comment by HackShitUp** [comment]
@flovilmart I actually resolved it about 6 hours ago. For anyone else who might have issues with this in the future, I resolved it by doing the following:

For video files to upload properly, the .ebextensions folder must contain a files.config file that sets the client's max upload size (https://github.com/parse-community/parse-server/issues/2671). When the files are zipped (via MAC OS), it also creates an unwanted directory that AWS doesn't allow in YAML format. To resolve this, delete that extra directory (https://stackoverflow.com/questions/34373107/deploy-to-aws-eb-failing-because-of-yaml-error-in-python-config) after compressing the file.
